common/include/common/common_utils.hpp

```diff
@@ -329,6 +331,45 @@ static inline std::vector<spr_param_t<scalar_t>> get_spr_params(Args& args) {
   }
 }
 
+/**
+ * @fn get_spr2_params
+ * @brief Returns a vector containing the spr2 benchmark parameters, either
+ * read from a file according to the command-line args, or the default ones.
+ */
+template <typename scalar_t>
+static inline std::vector<spr2_param_t<scalar_t>> get_spr2_params(Args& args) {
+  if (args.csv_param.empty()) {
+    warning_no_csv();
+    std::vector<spr2_param_t<scalar_t>> spr2_default;
+    constexpr index_t dmin = 64, dmax = 1024;
+    const index_t incX = 1;
+    const index_t incY = 1;
+    const scalar_t alpha = 1;
+    for (std::string uplo : {"u", "l"}) {
+      for (index_t m = dmin; m <= dmax; m *= 2) {
+        spr2_default.push_back(std::make_tuple(uplo, m, alpha, incX, incY));
+      }
+    }
+    return spr2_default;
+  } else {
+    return parse_csv_file<spr2_param_t<scalar_t>>(
+        args.csv_param, [&](std::vector<std::string>& v) {
+          if (v.size() != 5) {
+            throw std::runtime_error(
+                "invalid number of parameters (4 expected)");
```

Comment:
Thanks for your patch, @s-Nick.

```suggestion
                "invalid number of parameters (5 expected)");
```